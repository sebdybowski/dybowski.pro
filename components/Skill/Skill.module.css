@property --p {
    syntax: '<number>';
    inherits: true;
    initial-value: 0;
}

.pie {
    --p: 20;
    --b: 22px;
    --c: var(--primary-color);
    --w: 150px;

    width: var(--w);
    aspect-ratio: 1;
    position: relative;
    display: inline-grid;
    margin: 5px;
    place-content: center;
    font-size: 25px;
    font-weight: bold;
    font-family: sans-serif;
    transition: transform 0.15s ease-in-out;
}

.pie:hover {
    cursor: pointer;
    transform: scale(1.1);
    transition: transform 0.15s ease-in-out;
}

.pie:before {
    content: "";
    position: absolute;
    border-radius: 50%;
}

.pie::after {
    content: "";
    inset: 0;
    position: absolute;
    border-radius: 50%;
    border: 0.5rem solid rgba(0, 0, 0, 0.2);
    width: calc(100% - 1rem);
    height: calc(100% - 1rem);
    z-index: 0;
}

.pie:before {
    inset: 0;
    background:
        conic-gradient(var(--c) calc(var(--p) * 1%), #0000 0);
    mask: radial-gradient(farthest-side, #0000 calc(99% - var(--b)), #000 calc(100% - var(--b)));
    position: absolute;
    z-index: 1;
}

.animate {
    animation: p 1s .5s both;
}

.no-round:before {
    background-size: 0 0, auto;
}

.no-round:after {
    content: none;
}

@keyframes p {
    from {
        --p: 0
    }
}

.Skill {
    display: flex;
    justify-content: center;
    align-items: center;
    height: 15vw;
    width: 15vw;
    border-radius: 15vw;
    margin: 4rem 2rem;
}

.Skill>.Label {
    font-size: 2rem;
}