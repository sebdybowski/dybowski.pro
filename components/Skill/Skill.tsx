import clsx from 'clsx';
import React, { FC, MouseEventHandler } from 'react';

import styles from './Skill.module.scss';

interface SkillProps {
  label: string,
  percent: number,
  onClick: MouseEventHandler<HTMLButtonElement>
}

const Skill: FC<SkillProps> = ({ label, percent, onClick }) => (
  <button onClick={onClick} data-testid="Skill">
    {/* @ts-ignore */}
    <div className={clsx(styles.Skill, styles.pie)} style={{ ['--p']: percent }}>
      <span className={styles.Label}>{label}</span>
      <span className={styles.Percentage}>{`${percent}%`}</span>
    </div>
  </button>
);

export default Skill;
