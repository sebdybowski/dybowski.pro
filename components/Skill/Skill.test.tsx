import React from 'react';
import { render } from '@testing-library/react';

import '@testing-library/jest-dom/extend-expect';
import Skill from './Skill';

describe('<Skill />', () => {
  test('it should mount', () => {
    const { asFragment } = render(<Skill label='moonwalk' percent={100} />);
    
    expect(asFragment()).toMatchSnapshot();
  });
});
