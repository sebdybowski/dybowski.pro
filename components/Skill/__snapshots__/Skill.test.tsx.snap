// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Skill /> it should mount 1`] = `
<DocumentFragment>
  <button>
    <div
      class="Skill pie"
      data-testid="Skill"
      style="--p: 100;"
    >
      <span
        class="Label"
      >
        moonwalk
      </span>
      <span
        class="Percentage"
      >
        100%
      </span>
    </div>
  </button>
</DocumentFragment>
`;
